include/openssl/ssl.h

```diff
@@ -250,6 +250,31 @@ typedef int (*tls_session_secret_cb_fn) (SSL *s, void *secret,
                                          STACK_OF(SSL_CIPHER) *peer_ciphers,
                                          const SSL_CIPHER **cipher, void *arg);
 
+/* Extension context codes */
+/* This extension is only allowed in TLS */
+#define SSL_EXT_TLS_ONLY                        0x0001
+/* This extension is only allowed in DTLS */
+#define SSL_EXT_DTLS_ONLY                       0x0002
+/* Some extensions may be allowed in DTLS but we don't implement them for it */
+#define SSL_EXT_TLS_IMPLEMENTATION_ONLY         0x0004
```

Comment:
Well I kind of expect DTLSv1.3 to come along before 10 years is up! :-)

WRT the name its tempting to write an essay in the names for these flags trying to explain it - but you're never going to cover everything just in the name. Really the name should just give you a hint/reminder as to what it does - and look in the docs for the full story.